core/src/main/java/org/elasticsearch/cluster/routing/UnassignedInfo.java

```diff
@@ -117,52 +117,76 @@
      * Note, ordering of the enum is important, make sure to add new values
      * at the end and handle version serialization properly.
      */
-    public enum AllocationAttempt implements Writeable {
+    public enum AllocationStatus implements Writeable {
         /**
          * The shard was denied allocation to a node because the allocation deciders all returned a NO decision
          */
-        DENIED_BY_ALLOCATION_DECIDERS,
+        DECIDERS_NO((byte) 0),
         /**
          * The shard was denied allocation to a node because there were no valid shard copies found for it;
          * this can happen on node restart with gateway allocation
          */
-        NO_VALID_SHARD_COPY,
+        NO_VALID_SHARD_COPY((byte) 1),
         /**
          * The allocation attempt was throttled on the shard by the allocation deciders
          */
-        THROTTLED_BY_ALLOCATION_DECIDERS,
+        DECIDERS_THROTTLED((byte) 2),
         /**
          * Waiting on getting shard data from all nodes before making a decision about where to allocate the shard
          */
-        FETCHING_SHARD_DATA,
+        FETCHING_SHARD_DATA((byte) 3),
         /**
          * Allocation decision has been delayed
          */
-        DELAYED_ALLOCATION,
+        DELAYED_ALLOCATION((byte) 4),
         /**
          * No allocation attempt has been made yet
          */
-        NO_ATTEMPT;
+        NO_ATTEMPT((byte) 5);
+
+        private final byte value;
+
+        AllocationStatus(byte value) {
+            this.value = value;
+        }
+
+        // package private for testing
+        byte getValue() {
+            return value;
+        }
 
         @Override
         public void writeTo(StreamOutput out) throws IOException {
-            out.writeByte((byte) ordinal());
+            out.writeByte(value);
         }
 
-        public static AllocationAttempt readFrom(StreamInput in) throws IOException {
-            return values()[(int) in.readByte()];
+        public static AllocationStatus readFrom(StreamInput in) throws IOException {
+            byte v = in.readByte();
+            switch (v) {
+                case 0:
+                    return DECIDERS_NO;
+                case 1:
+                    return NO_VALID_SHARD_COPY;
+                case 2:
+                    return DECIDERS_THROTTLED;
+                case 3:
+                    return FETCHING_SHARD_DATA;
+                case 4:
+                    return DELAYED_ALLOCATION;
+                case 5:
+                    return NO_ATTEMPT;
+                default:
+                    throw new IllegalArgumentException("Unknown AllocationStatus value [" + v + "]");
+            }
```

Comment:
for `toXContent` and `toString()`, we can also provide a method here that gives a nice description string .